core/network/source.go

```diff
@@ -29,7 +44,7 @@ type ConfigSource interface {
 
 	// InterfaceAddresses returns information about all addresses
 	// assigned to the network interface with the given name.
-	InterfaceAddresses(name string) ([]net.Addr, error)
+	InterfaceAddresses(name string) ([]ConfigSourceAddr, error)
```

Comment:
I don't think this method is actually required, because when we call it, we have already acquired the actual interfaces, from which we can get the addresses directly. I will reorganise in following patches.